contracts/staking/TokenStaking.sol

```diff
@@ -0,0 +1,1236 @@
+                    gas: GAS_LIMIT_AUTHORIZATION_DECREASE
+                }(operatorAddress, authorization.authorized - totalStake)
+            {} catch {}
+            authorization.authorized = totalStake;
+            if (authorization.deauthorizing > totalStake) {
+                authorization.deauthorizing = totalStake;
+            }
+        }
+    }
+
+    /// @notice Convert amount from T to Keep and call `seize` in Keep staking contract.
+    ///         Returns remainder of slashing amount in T
+    function seizeKeep(
+        OperatorInfo storage operator,
+        address operatorAddress,
+        uint96 tAmountToSlash,
+        uint256 rewardMultiplier
+    ) internal returns (uint96) {
+        uint96 tPenalty;
+        if (tAmountToSlash <= operator.keepStake) {
+            tPenalty = tAmountToSlash;
+            operator.keepStake -= tAmountToSlash;
+        } else {
+            tPenalty = operator.keepStake;
+            operator.keepStake = 0;
+        }
+
+        (uint256 keepPenalty, uint96 tRemainder) = tToKeep(tPenalty);
+        operator.keepStake += tRemainder;
+        tAmountToSlash -= tPenalty - tRemainder;
+
+        address[] memory operatorWrapper = new address[](1);
+        operatorWrapper[0] = operatorAddress;
+        keepStakingContract.seize(
+            keepPenalty,
+            rewardMultiplier,
+            msg.sender,
+            operatorWrapper
+        );
+        return tAmountToSlash;
+    }
+
+    /// @notice Convert amount from T to NU and call `slashStaker` in NuCypher staking contract.
+    ///         Returns remainder of slashing amount in T
+    function seizeNu(
+        OperatorInfo storage operator,
+        uint96 tAmountToSlash,
+        uint256 rewardMultiplier
+    ) internal returns (uint96) {
+        uint96 tPenalty;
+        if (tAmountToSlash <= operator.nuStake) {
+            operator.nuStake -= tAmountToSlash;
+            tPenalty = tAmountToSlash;
+        } else {
+            tPenalty = operator.nuStake;
+            operator.nuStake = 0;
+        }
+
+        (uint256 nuPenalty, uint96 tRemainder) = tToNu(tPenalty);
+        operator.nuStake += tRemainder;
+        tAmountToSlash -= tPenalty - tRemainder;
+
+        uint256 nuReward = nuPenalty.percent(SLASHING_REWARD_PERCENT).percent(
+            rewardMultiplier
+        );
+        nucypherStakingContract.slashStaker(
+            operator.owner,
+            nuPenalty,
+            msg.sender,
+            nuReward
+        );
+        return tAmountToSlash;
+    }
+
+    /// @notice Returns amount of Keep stake in the Keep staking contract for the specified operator.
+    ///         Resulting value in T denomination
+    function getKeepAmountInT(address operator) internal view returns (uint96) {
+        (, uint256 createdAt, ) = keepStakingContract.getDelegationInfo(
+            operator
+        );
+        require(createdAt != 0, "Nothing to sync");
```

Comment:
What is the reasoning behind this check? If the eligible stake is > 0 it means there is KEEP staked and T staking contract is authorized as an operator contract. I think it should be enough but obviously possible I could have missed something.

Reply:
Agree, this check is not necessary. Just for case when no eligible stake on Keep side but staker want to have this operator in T staking contract. Want to hear your opinion on that

Reply:
Just removing this line will break `stakeKeep`. I guess you mean add `require(tAmount > 0)` instead of removed line for both `stakeKeep` and `topUpKeep`, right?

Reply:
Yeah, that's what I meant. Sorry, I was not clear enough.

If `KeepStake` was always resolving all owners (this is not possible - some of them are conflicting between two staking contract versions we had), the current approach would have the advantage of being able to reuse undelegated KEEP operator account and operating in T network. But... I think this is quite complex concept to understand, a bit clumsy and `KeepStake` does not support it.

Reply:
Ok, changed
✔️